internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_cloud_resource.go

```diff
@@ -0,0 +1,209 @@
+	Tags                                                   map[string]string `tfschema:"tags"`
+}
+
+var _ sdk.Resource = SystemCenterVirtualMachineManagerCloudResource{}
+var _ sdk.ResourceWithUpdate = SystemCenterVirtualMachineManagerCloudResource{}
+
+type SystemCenterVirtualMachineManagerCloudResource struct{}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) ModelObject() interface{} {
+	return &SystemCenterVirtualMachineManagerCloudModel{}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return clouds.ValidateCloudID
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) ResourceType() string {
+	return "azurerm_system_center_virtual_machine_manager_cloud"
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SystemCenterVirtualMachineManagerCloudName,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"custom_location_id": commonschema.ResourceIDReferenceRequiredForceNew(&customlocations.CustomLocationId{}),
+
+		"system_center_virtual_machine_manager_server_inventory_item_id": commonschema.ResourceIDReferenceRequiredForceNew(&inventoryitems.InventoryItemId{}),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.SystemCenterVirtualMachineManager.Clouds
+
+			var model SystemCenterVirtualMachineManagerCloudModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			scvmmServerInventoryItemId, err := inventoryitems.ParseInventoryItemIDInsensitively(model.SystemCenterVirtualMachineManagerServerInventoryItemId)
+			if err != nil {
+				return err
+			}
+
+			id := clouds.NewCloudID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := clouds.Cloud{
+				Location: location.Normalize(model.Location),
+				ExtendedLocation: clouds.ExtendedLocation{
+					Type: utils.String("customLocation"),
```

Comment:
please also update other places to keep consistent
```suggestion
					Type: pointer.To("customLocation"),
```

Reply:
Updated